Adding more specific info on signing CSR using 3rd-party CA for SAML certs

motikagan-okta opened the pull request:
## Description:
- **What's changed?** In the guide on signing SAML certificates, adding some more content, suggested by Brian, on signing the CSR using popular CAs.
- **Is this PR related to a Monolith release?** No.
### Resolves:

* [OKTA-309083](https://oktainc.atlassian.net/browse/OKTA-309083)

IanHakes-okta commented on packages/@okta/vuepress-site/docs/guides/sign-your-own-saml-csr/sign-the-csr/index.md:
```diff
@@ -2,7 +2,22 @@
 ---
 
-Follow the third-party process that your company uses to sign the CSR. You can't move to step four until you complete the process because the signed CSR is required to continue.
+Follow the third-party Certificate Authority (CA) process that your company uses to sign the CSR. You cannot proceed to the next step until your certificate is signed using this process.
```
cannot => can't

motikagan-okta replied:
made the change

IanHakes-okta commented on packages/@okta/vuepress-site/docs/guides/sign-your-own-saml-csr/sign-the-csr/index.md:
```diff
@@ -2,7 +2,22 @@
+
+> **Note:** There is a cost associated with SSL certificates being signed by a third-party CA.
+
+The certificate authority you choose will provide instructions for how to upload the CSR you generated in the previous steps.
```
for how => on how

motikagan-okta replied:
made the change

IanHakes-okta commented on packages/@okta/vuepress-site/docs/guides/sign-your-own-saml-csr/sign-the-csr/index.md:
```diff
@@ -2,7 +2,22 @@
+Follow the third-party Certificate Authority (CA) process that your company uses to sign the CSR. You cannot proceed to the next step until your certificate is signed using this process.
+
+The following is a list of popular third-party CA providers that can be used to sign the CSR:
+ - Comodo SSL
```
Should these be live links to the sites in question?

motikagan-okta replied:
I don't know if we want to be responsible for maintaining that, so safer not to.

IanHakes-okta commented on packages/@okta/vuepress-site/docs/guides/sign-your-own-saml-csr/sign-the-csr/index.md:
```diff
@@ -2,7 +2,22 @@
+The following is a list of popular third-party CA providers that can be used to sign the CSR:
+ - Comodo SSL
+ - DigiCert
+ - GoDaddy
```
GoDaddy should be after GlobalSign in alphabetical order

motikagan-okta replied:
made the change

cwu-okta commented on packages/@okta/vuepress-site/docs/guides/sign-your-own-saml-csr/sign-the-csr/index.md:
```diff
@@ -2,7 +2,22 @@
-Follow the third-party process that your company uses to sign the CSR. You can't move to step four until you complete the process because the signed CSR is required to continue.
+Follow the third-party Certificate Authority (CA) process that your company uses to sign the CSR. You can't proceed to the next step until your certificate is signed using this process.
+
+The following is a list of popular third-party CA providers that can be used to sign the CSR:
```
I don't think we want to list this, do we? I allowed this in the SFDC KB article but I'm not sure if we want to put it on the KB site. I defer to PM though if they want to add this. It will be the first time we put CA vendors on the DOC site.

motikagan-okta replied:
@venkatviswanathan-okta, @chrismorris-okta, should we be listing names of specific CAs in d.o.c. content?

venkatviswanathan-okta replied:
I don't think we need to list CAs in our doc. We don't need to be opinionated on what CA a customer should use. As long as you generate the CSR from Okta and get it signed by a CA, we should be good.

chrismorris-okta replied:
I have no problem listing this out if it useful. I've consistently heard from the developers I spoke with that they want us to provide more guidance and best practices on what they should do and how. Also, we should always create docs for the lowest common denominator. e.g. A front-end developer may not be familiar with this process at all. Just curious, have we gotten any feedback from support or the feedback form regarding this, or is this something we we are proposing on our own? 

My only concern is keeping this section up to date. I don't think we should embed the instructions in our docs, but just list the top 3-4 CAs and link out to their docs for the steps.

venkatviswanathan-okta replied:
`e.g. A front-end developer may not be familiar with this process at all.` - This is a SAML App feature and the target persona is an Enterprise IT Administrator. Companies use a wide variety of certificates Authority.

cwu-okta commented:
On the guide's step to update the app with the new cert we should add a warning to indicate it means SSO will stop working. For SAML, we need to update both the IdP (Okta app) and SP with the new cert before sso will work. While in a partial updated state SSO will not work.

susanharper-okta commented on packages/@okta/vuepress-site/docs/guides/sign-your-own-saml-csr/sign-the-csr/index.md:
```diff
@@ -2,7 +2,22 @@
+
+> **Note:** There is a cost associated with SSL certificates being signed by a third-party CA.
+
+The certificate authority you choose will provide instructions on how to upload the CSR you generated in the previous steps.
```
The certificate authority that you choose provides instructions on how to upload the CSR that you generated in the previous steps.

cwu-okta replied:
Nit: we mention CA above and certificate authority here. Should we define earlier that CA = certificate authority? Or, if not, mention here `certificate authority (CA)`

venkatviswanathan-okta replied:
I recommend using certificate authority (CA)

motikagan-okta commented:
> On the guide's step to update the app with the new cert we should add a warning to indicate it means SSO will stop working. For SAML, we need to update both the IdP (Okta app) and SP with the new cert before sso will work. While in a partial updated state SSO will not work.

Added a sentence to the "Sign the CSR" step, saying: "After you update the key credential, your users can't access the SAML app until you upload the new certificate to the Service Provider (SP)."

Also added a note to the overview at the beginning of the guide, saying: "After you update the key credential, your users can't access the SAML app until you upload the new certificate to the ISV."
